staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -537,7 +542,7 @@ func DefaultBuildHandlerChain(apiHandler http.Handler, c *Config) http.Handler {
 	handler = genericfilters.WithTimeoutForNonLongRunningRequests(handler, c.RequestContextMapper, c.LongRunningFunc, c.RequestTimeout)
 	handler = genericapifilters.WithRequestInfo(handler, c.RequestInfoResolver, c.RequestContextMapper)
 	handler = apirequest.WithRequestContext(handler, c.RequestContextMapper)
-	handler = genericfilters.WithPanicRecovery(handler)
+	handler = genericfilters.WithPanicRecovery(handler, c.HandlerChainWaitGroup)
```

Comment:
I tend to prefer another filter. The HandlerChainWaitGroup logic is orthogonal to panic recovery.

Reply:
OK. this code is all nit. I just show you my implementation. This way is viable.